packages/paperstore/tests/test_sqlite_backend.py

```
@@ -103,10 +139,228 @@ def test_upsert_year_and_list_papers(store: SqliteBackend):
+    source_path = store.put_source("P1", b"bytes", suffix=".pdf")
+    md_path = store.write_paper_md("P1", "# body\n")
+    store.write_meta_json("P1", {"title": "T", "year": "2026"})
+    meta = store.get_meta("P1")
+    assert meta["title"] == "T"
+    assert meta["source_file"] == str(source_path)
+    assert meta["markdown_path"] == str(md_path)
+
+
+def test_write_meta_json_can_set_source_file_when_provided(store: SqliteBackend):
+    """write_meta_json still writes columns the caller explicitly supplies."""
+    store.write_meta_json("P1", {"title": "T", "source_file": "/tmp/explicit.pdf"})
```

Ruff (0.15.12)

[error] Probable insecure usage of temporary file or directory: "/tmp/explicit.pdf"

(S108)